synapse/handlers/identity.py

```diff
@@ -59,21 +61,23 @@ def __init__(self, hs):
         self.federation_http_client = hs.get_http_client()
         self.hs = hs
 
-    async def threepid_from_creds(self, id_server, creds):
+    async def threepid_from_creds(
+        self, id_server: str, creds: Dict[str, str]
+    ) -> Dict[str, Optional[Union[str, int]]]:
```

Comment:
I think it should be
```suggestion
    ) -> Optional[Dict[str, Union[str, int]]]:
```

Or possibly just
```suggestion
    ) -> Optional[JsonDict]:
```

Reply:
Hmm.... yeah the old type was `Deferred[dict[str,str|int]|None]` I think I failed to follow the nested braces properly.

`JsonDict` sounds like a good improvement here. 😄